Formula/v/vineflower.rb

```diff
@@ -0,0 +1,33 @@
+class Vineflower < Formula
+  desc "Java decompiler"
+  homepage "https://vineflower.org/"
+  url "https://github.com/Vineflower/vineflower/archive/refs/tags/1.11.1.tar.gz"
+  sha256 "104df4042023190416bff2ad0e2386ddf7669c6aa585f19b9d965a7f5ca5132b"
+  license "Apache-2.0"
+
+  depends_on "openjdk@21"
+
+  def install
+    system "./gradlew", "build", "-x", "test"
+    jar = Dir["build/libs/vineflower-#{version}+local.jar"].first
+    libexec.install jar => "vineflower.jar"
+
+    (bin / "vineflower").write <<~SH
+      #!/bin/sh
+      exec "#{Formula["openjdk@21"].opt_bin}/java" -jar "#{libexec}/vineflower.jar" "$@"
+    SH
```

Comment:
I believe you can use `write_jar_script` here instead:
```suggestion
    bin.write_jar_script libexec/"vineflower.jar", "vineflower", java_version: "21"
```